src/psyclone/psyir/transformations/intrinsics/matmul2code_trans.py

```diff
@@ -106,29 +106,33 @@ def _get_array_bound(array, index):
             f"Unsupported index type found for array '{array.name}': "
             f"{err}") from err
 
+    dim_index = index + 1  # The Fortran dim argument is 1-indexed
```

Comment:
This worries me a bit but probably only highlights an existing issue: we're dealing with PSyIR here. The fact that Fortran is 1-indexed should really be something that the Fortran backend deals with? Or is the PSyIR 1-indexed (I can't remember)?

Reply:
We are dealing with the dim argument that is passed to the intrinsic. It would be very tricky to make index conversions in the backend. I think setting them up appropriately in the transformations like we are doing here is the best we can do.

Reply:
Yes, agreed.